src/a2a/server/apps/rest/rest_app.py

```diff
@@ -0,0 +1,237 @@
+    ):
+        """Initializes the RESTApplication.
+
+        Args:
+            agent_card: The AgentCard describing the agent's capabilities.
+            http_handler: The handler instance responsible for processing A2A
+              requests via http.
+            context_builder: The CallContextBuilder used to construct the
+              ServerCallContext passed to the http_handler. If None, no
+              ServerCallContext is passed.
+        """
+        self.agent_card = agent_card
+        self.handler = RESTHandler(
+            agent_card=agent_card, request_handler=http_handler
+        )
+        self._context_builder = context_builder or DefaultCallContextBuilder()
+
+    def _generate_error_response(self, error: A2AError) -> JSONResponse:
+        """Creates a JSONResponse for an error.
+
+        Logs the error based on its type.
+
+        Args:
+            error: The Error object.
+
+        Returns:
+            A `JSONResponse` object formatted as a JSON error response.
+        """
+        log_level = (
+            logging.ERROR
+            if isinstance(error, InternalError)
+            else logging.WARNING
+        )
+        logger.log(
+            log_level,
+            'Request Error: '
+            f"Code={error.root.code}, Message='{error.root.message}'"
+            f'{", Data=" + str(error.root.data) if error.root.data else ""}',
+        )
+        return JSONResponse(
+            f'{{"message": "{error.root.message}"}}',
+            status_code=500,
+        )
+
+    def _handle_error(self, error: Exception) -> JSONResponse:
+        traceback.print_exc()
+        if isinstance(error, MethodNotImplementedError):
+            return self._generate_error_response(
+                A2AError(UnsupportedOperationError(message=error.message))
+            )
+        if isinstance(error, json.decoder.JSONDecodeError):
+            return self._generate_error_response(
+                A2AError(JSONParseError(message=str(error)))
+            )
+        if isinstance(error, ValidationError):
+            return self._generate_error_response(
+                A2AError(InvalidRequestError(data=json.loads(error.json()))),
+            )
+        logger.error(f'Unhandled exception: {error}')
+        return self._generate_error_response(
+            A2AError(InternalError(message=str(error)))
+        )
+
+    async def _handle_request(
+        self,
+        method: Callable[[Request, ServerCallContext], Awaitable[str]],
+        request: Request,
+    ) -> JSONResponse:
+        try:
+            call_context = self._context_builder.build(request)
+            response = await method(request, call_context)
+            return JSONResponse(content=response)
+        except Exception as e:
+            return self._handle_error(e)
+
+    async def _handle_streaming_request(
+        self,
+        method: Callable[[Request, ServerCallContext], AsyncIterator[str]],
```

Comment:
Typing: should be AsyncIterator[Any]. This assumes that each handler produces a string, but we want to allow the data to be anything. Alternatively, might want to be dict[str, Any] to enforce that it's at least an object.